plugins/speculation-rules/settings.php

```diff
@@ -281,9 +309,39 @@ function plsr_render_settings_field( array $args ): void {
 		<?php endforeach; ?>
 
 		<p class="description" style="max-width: 800px;">
-			<?php echo esc_html( $args['description'] ); ?>
+			<?php
+			echo wp_kses(
+				$args['description'],
+				array(
+					'a' => array(
+						'href'   => array(),
+						'target' => array(),
+					),
+				)
+			);
+			?>
 		</p>
 	</fieldset>
+	
+	<?php if ( 'authentication' === $args['field'] && ! $has_object_cache ) : ?>
+		<script>
+		( function () {
+			const authRadios = document.querySelectorAll( '.plsr-auth-radio' );
+			const warningDiv = document.getElementById( 'plsr-auth-warning' );
+			const authenticatedOptions = <?php echo wp_json_encode( $authenticated_options ); ?>;
+			
+			if ( ! authRadios.length || ! warningDiv ) {
+				return;
+			}
+			
+			authRadios.forEach( function ( radio ) {
+				radio.addEventListener( 'change', function ( e ) {
+					warningDiv.hidden = ! authenticatedOptions.includes( e.target.value );
+				} );
+			} );
+		} )();
+		</script>
```

Comment:
This should use `wp_print_inline_script_tag()` along with `type="module"`. This will ensure it is compatible with CSP and the use of script modules will eliminate the need for the IIFE wrapper, since modules do not pollute the global scope.

Reply:
Related:
- https://github.com/WordPress/WordPress-Coding-Standards/issues/63